common/containers/Tabs/SendTransaction/index.tsx

```diff
@@ -1,775 +1,52 @@
-                            <textarea
-                              className="form-control"
-                              value={transaction.rawTx}
-                              rows={4}
-                              readOnly={true}
-                            />
-                          </div>
-                          <div className="col-sm-6">
-                            <label>{translate('SEND_signed')}</label>
-                            <textarea
-                              className="form-control"
-                              value={transaction.signedTx}
-                              rows={4}
-                              readOnly={true}
-                            />
-                            {offline && (
-                              <p>
-                                To broadcast this transaction, paste the above
-                                into{' '}
-                                <a href="https://myetherwallet.com/pushTx">
-                                  {' '}
-                                  myetherwallet.com/pushTx
-                                </a>{' '}
-                                or{' '}
-                                <a href="https://etherscan.io/pushTx">
-                                  {' '}
-                                  etherscan.io/pushTx
-                                </a>
-                              </p>
-                            )}
-                          </div>
-                        </div>
-
-                        {!offline && (
-                          <div className="row form-group">
-                            <div className="col-xs-12">
-                              <button
-                                className="btn btn-primary btn-block"
-                                disabled={!this.state.transaction}
-                                onClick={this.openTxModal}
-                              >
-                                {translate('SEND_trans')}
-                              </button>
-                            </div>
-                          </div>
-                        )}
-                      </div>
-                    )}
-                  </div>
-                </main>
-              )}
-
-            {unlocked &&
-              ((offline || forceOffline) && isWeb3Wallet) && (
-                <main className="col-sm-8">
-                  <div className="Tab-content-pane">
-                    <h4>Sorry...</h4>
-                    <p>
-                      MetaMask / Mist wallets are not available in offline mode.
-                    </p>
-                  </div>
-                </main>
-              )}
-
-            {/* Sidebar */}
-            {unlocked && (
-              <section className="col-sm-4">
-                <BalanceSidebar />
-              </section>
-            )}
+            <Fields />
+            <UnavailableWallets />
+            <SideBar />
```

Comment:
Is there a reason these 3 components aren't simplified by putting them inside of `<Wallet withWallet={() => ...`? None of them should render unless there's a wallet available.

Reply:
I'm just now realizing that this renderCb is actually just a selector, it doesn't only render when the wallet has been unlocked. I find that a little confusing, I'd assume `withWallet` would mean 'with a wallet unlocked', not 'with the wallet state provided (which might not have a wallet instance.)'. At that point, `connect`ing this component seems a lot simpler.